fl4health/clients/basic_client.py

```diff
@@ -332,30 +322,40 @@ def get_parameter_exchanger(self, config: Config) -> ParameterExchanger:
         """
         return FullParameterExchanger()
 
-    def predict(self, input: torch.Tensor) -> Dict[str, torch.Tensor]:
+    def predict(self, input: torch.Tensor) -> Tuple[Dict[str, torch.Tensor], Dict[str, torch.Tensor]]:
         """
-        Return dict of str and torch.Tensor containing predictions when given input.
-        In the default case, the dict has a single item with key prediction.
-        In more complicated approaches such as APFL, the dict has as many items as prediction types
-        User can override for more complex logic.
+        Computes the prediction(s) (and potentially features) of the model(s) given the input.
+
+        Args:
+            input (torch.Tensor): Inputs to be fed into the model.
+
+        Returns:
+            Tuple[Dict[str, torch.Tensor], Dict[str, torch.Tensor]]: A tuple in which the first element
+            contains predictions indexed by name and the second element contains intermediate activations
```

Comment:
I see you added a comment in the compute loss function about anything stored in preds being used to compute metrics, but I think we should also put that in the comment for the `predict` function since that's where people will define what's in the predictions dictionary. So they "know" this before potentially stuffing extra things into that dictionary.

Reply:
Good call!